Greet-Dough/src/main/java/store/PostStoreImpl.java

```diff
@@ -0,0 +1,44 @@
+package store;
+
+import model.Post;
+import java.util.HashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class PostStoreImpl implements PostStore {
+
+    private AtomicInteger freeIDs;
+    private final HashMap<Integer, Post> posts;
+
+    public PostStoreImpl() {
+        this(0);
+    }
+
+    public PostStoreImpl( int start ) {
+
+        this.freeIDs = new AtomicInteger(start);
+        this.posts = new HashMap<>();
+
+    }
+
+    @Override
+    public int getFreeID() {
```

Comment:
you're overriding this from the `TrackerID`. might as well just have each store maintain its own AtomicInteger/HashMap, and then you don't need the TrackerID concept necessarily